copyright.js

```diff
@@ -1,5 +1,5 @@
 /**
- * © 2017 Liferay, Inc. <https://liferay.com>
+ * © <%= YEAR %> Liferay, Inc. <https://liferay.com>
```

Comment:
(Just thinking out loud here...)

This will default to current YEAR, right? According to the various places where we have written about this:

* [This loop post](https://loop.liferay.com/es/home/-/loop/feed/12995353) says it should be "the year that was entered when the file was created".
* [This wiki page](https://grow.liferay.com/share/Liferay+Policy+for+Marking+Our+Own+Source+Code) says it should be "year when this file was first created/published".
* [This comment](https://github.com/liferay/clay/pull/1451#issuecomment-456384580) says "the year to whatever year that file was in (if that’s too much work, 2014 as the year of the first commit should do)"

It's that last rule that made me put 2017 here, because 2017 was the year of the first commit to this repo. If we change this to `YEAR`, then it will do the right thing for *new* files, because they'll get the whatever the current year is at the time the file is created. I think that's right for this project, but only because all the files currently have the headers they need.

But for projects that have pre-existing code, *if* they copy this as an example, then they'd get "2019" added to all their files, including things created in prior years, and that would be wrong (for the same reason that 2017 would be wrong). That's a big "if", but I wonder if there is a way to make this more "fool-proof" for such projects. Any ideas?




we would want this to be "year of first commit" of project

Reply:
Good point about this, in fact I would expect the plugin would do this for us, according to the date of creation of the file, check the valid year for such. But it seems he does not do that 😞. That would be the only idea I would have... I do not know if we can do something like that on our side.

Reply:
Yeah I think the problem is that Git doesn't manage creation dates, so if I clone the repo and then run the lint, "2019" will be used as creation date. Given that Git is used so pervasively, it's a pretty good argument against implementing the functionality in the plug-in.

Reply:
Yeah, I do not know if it would help a lot, but maybe we can add warnings about using copyright properly by referring to posts. In the documentation here or README.md, where to become more visible.